contracts/account/extensions/AccountERC6900.sol

```diff
@@ -0,0 +1,494 @@
+        );
+        // The account MUST install all execution functions and set flags and fields as specified in the manifest.
+        ManifestExecutionFunction[] memory executionFunctions = manifest.executionFunctions;
+        for (uint256 i = 0; i < executionFunctions.length; i++) {
+            ManifestExecutionFunction memory executionFunction = executionFunctions[i];
+            bytes4 executionSelector = executionFunction.executionSelector;
+            Execution storage execution = _executions[executionSelector];
+            // An execution function selector MUST be unique in the account.
+            require(
+                execution.module == address(0),
+                ERC6900Utils.ERC6900AlreadyUsedModuleFunctionExecutionSelector(executionSelector)
+            );
+            // An execution function selector MUST not conflict with native ERC-4337 and ERC-6900 functions.
+            require(
+                IAccount.validateUserOp.selector != executionSelector, // TODO Check other ERC-4337 functions
+                ERC6900Utils.ERC6900ExecutionSelectorConflictingWithERC4337Function(module, executionSelector)
+            );
+            require(
+                IERC6900ModularAccount.execute.selector != executionSelector, // TODO Check other ERC-6900 functions
+                ERC6900Utils.ERC6900ExecutionSelectorConflictingWithERC6900Function(module, executionSelector)
+            );
+            execution.module = module;
+            execution.skipRuntimeValidation = executionFunction.skipRuntimeValidation;
+            execution.allowGlobalValidation = executionFunction.allowGlobalValidation;
+        }
+        // The account MUST add all execution hooks as specified in the manifest.
+        ManifestExecutionHook[] memory executionHooks = manifest.executionHooks;
+        for (uint256 i = 0; i < executionHooks.length; i++) {
+            ManifestExecutionHook memory executionHook = executionHooks[i];
+            bytes4 executionSelector = executionHook.executionSelector;
+            // module
+            uint32 entityId = executionHook.entityId;
+            bool isPreHook = executionHook.isPreHook;
+            bool isPostHook = executionHook.isPostHook;
+            Execution storage execution = _executions[executionSelector];
+            require(
+                execution.executionHooks.add(bytes32(abi.encodePacked(module, entityId, isPreHook, isPostHook))),
+                ERC6900Utils.ERC6900AlreadySetExecutionHookForExecution()
+            );
+        }
+        // The account SHOULD add all supported interfaces as specified in the manifest.
+        bytes4[] memory interfaceIds = manifest.interfaceIds;
+        for (uint256 i = 0; i < interfaceIds.length; i++) {
+            bytes4 interfaceId = interfaceIds[i];
+            require(!_interfaceIds[interfaceId], "Interface already set"); // TODO use custom error
+            _interfaceIds[interfaceId] = true;
```

Comment:
What if two execution functions add the same interface id? Are all situations not allowed?

Reply:
You are right, a same interface can be added multiple times as ref implem show, see [Count supported interfaces](https://github.com/OpenZeppelin/openzeppelin-community-contracts/pull/80/commits/21949b099c4c2a5c564a2a934cd6062599d554a0).